include/sleipnir/autodiff/Expression.hpp

```diff
@@ -299,11 +300,26 @@ struct SLEIPNIR_DLLEXPORT Expression {
   friend SLEIPNIR_DLLEXPORT sleipnir::IntrusiveSharedPtr<Expression> operator+(
       const sleipnir::IntrusiveSharedPtr<Expression>& lhs);
 
+  /**
+   * Breadth-first search (BFS) of the expression's computational tree. BFS is
+   * sed as opposed to a depth-first search (DFS) to avoid counting duplicate
+   * nodes multiple times. A list of nodes ordered from parent to child with
+   * no duplicates is generated.
+   * https://en.wikipedia.org/wiki/Breadth-first_search
+   */
+  std::vector<Expression*> GenerateBFS();
```

Comment:
If users aren't supposed to modify or access this vector directly, I suggest wrapping it in a type like BFSGraph and make Expression::UpdateGraph() a member function for it. Then its usage can be enforced through the API, and we can better document what the data type represents.

The type would take an Expression& in its constructor and update a private `std::vector<Expression*>`. The only public member function would be Update().